EmotiBitWiFi.cpp

```diff
@@ -76,26 +81,64 @@ uint8_t EmotiBitWiFi::begin(const String &ssid, const String &pass, uint8_t maxA
 			return wifiStatus;
 		}
 		Serial.print("Attempting to connect to SSID: ");
-		Serial.println(ssid);
+		Serial.println(credential.ssid);
 		// ToDo: Add WEP support
 		_wifiOff = false;
 		unsigned long beginDuration = millis();
-		if (pass.equals(""))
+		if (credential.pass.equals(""))
 		{
 			// assume open network if password is empty
-			wifiStatus = WiFi.begin(ssid.c_str());
+			wifiStatus = WiFi.begin(credential.ssid.c_str());
 		}
 		else
 		{
-			wifiStatus = WiFi.begin(ssid.c_str(), pass.c_str());
+			if (credential.userid != "")
+			{
+#ifdef ARDUINO_FEATHER_ESP32
+				// enterprise WiFi
+				String username = "";
+				if (credential.username.equals(""))
+				{
+					// If username is blank, user userid as user-name
+					username = credential.userid;
+				}
+				else
+				{
+					username = credential.username;
+				}
+				// uncomment for debugging
+				Serial.print("trying enterprise WiFi: "); 
+				Serial.print("-SSID:"); Serial.print(credential.ssid);
+				Serial.print(" -userid:"); Serial.print(credential.userid);
+				Serial.print(" -username:"); Serial.print(username);
+				Serial.print(" -pass:"); Serial.println(credential.pass);
+        WiFi.mode(WIFI_STA); //init wifi mode
+        esp_wifi_sta_wpa2_ent_set_identity((uint8_t *)credential.userid.c_str(), credential.userid.length());
+        esp_wifi_sta_wpa2_ent_set_username((uint8_t *)username.c_str(), username.length());
+        esp_wifi_sta_wpa2_ent_set_password((uint8_t *)credential.pass.c_str(), credential.pass.length());
+        esp_wifi_sta_wpa2_ent_enable();
+        wifiStatus = WiFi.begin(credential.ssid.c_str());
+				//WiFi.begin(credential.ssid.c_str(), WPA2_AUTH_PEAP, credential.userid.c_str(), username.c_str(), credential.pass.c_str());
+#else
+				// do nothing. Enterprise support only for ESP32
+				Serial.print("Skipping Enterprise SSID: "); Serial.println(credential.ssid);
+				Serial.println("Enterprise WiFi is supported only for ESP32");
+#endif
+			}
+			else
+			{
+				// personal WiFi
+				wifiStatus = WiFi.begin(credential.ssid.c_str(), credential.pass.c_str());
+			}
 		}
 		Serial.print("WiFi.begin() duration = ");
 		Serial.println(millis() - beginDuration);
 		wifiStatus = status();
 		_needsAdvertisingBegin = true;
-		while((wifiStatus == WL_IDLE_STATUS) && (millis() - beginDuration < attemptDelay)); // This is necessary for ESP32 unless callback is utilized
+		//while((wifiStatus == WL_IDLE_STATUS) && (millis() - beginDuration < attemptDelay)) // This is necessary for ESP32 unless callback is utilized
+    while((wifiStatus != WL_CONNECTED) && (millis() - beginDuration < attemptDelay)) // This is necessary for ESP32 unless callback is utilized
```

Comment:
```
Attempting to connect to SSID: personal-1
WiFi.begin() duration = 134
attemptDelay: 20000  -  WiFi.status() = 6  // Wifi status == WL_DISCONNECTED
attemptDelay: 20000  -  WiFi.status() = 1  // WiFi status changes to WL_NO_SSID_AVAIL
WiFi.status() = 1, total duration = 4136   // This WL status makes it jump out of the while(wait for attempDelay) before 20secs
```

Therefore, the conditional in the `while` has been changed to make sure the MCU waits for `attemptDelay` seconds before moving on.

Reply:
Better to move on faster i guess.